### PR TITLE
Module: Mastodon

### PR DESCRIPTION
**Pull request by ifarzana**
- Relates to  https://github.com/xibosignage/xibo/issues/3042

We could not retrieve location for Mastodon topics 

Widget properties that could not be added as they don't support the following as filter
- Language
- Type (i.e., recent, popular topics)

These are the links we could retrieve data from
- https://mastodon.social/api/v1/timelines/public
- https://mastodon.social/api/v1/timelines/tag (you must provide tag to see results)

### CODE REVIEW

File: modules/mastodon.xml [hunk 1]
```diff
@@ -0,0 +1,167 @@
...
+            <rule>required</rule>
+        </property>
+        <property id="tag" type="text">
+            <title>Tag</title>
+            <helpText>Tag. You can test your search tag in the mastodon URL first.</helpText>
+            <default></default>
+        </property>
+        <property id="numItems" type="number">
+            <title>Count</title>
+            <helpText>The number of Tweets to return (default = 15).</helpText>
```
Comment (dasgarner) [inline modules/mastodon.xml:50]: Should we call this "posts" ?

File: modules/mastodon.xml
Comment (dasgarner) [inline modules/mastodon.xml:1]: Can we make sure we remove the word "Tweet" and replace it with whatever is common parlance for Mastodon?

File: lib/Widget/MastodonProvider.php [hunk 2]
```diff
@@ -0,0 +1,147 @@
...
+            }
+            $response = $dataProvider
+                ->getGuzzleClient($httpOptions)
+                ->get($uri);
+
+            $result = json_decode($response->getBody()->getContents(), true);
+
+            $this->getLog()->debug('Mastodon: count: ' . count($result));
+
+            // TODO when should we expire the media, we dont have any settings
```
Comment (dasgarner) [inline lib/Widget/MastodonProvider.php:70]: We should add this as an option to settings

**Review by dasgarner — COMMENTED** [review]
This looks like a good start! A few things to fix up and then we can get it into test I think

File: modules/mastodon.xml
Comment (maurofmferrao) [reply modules/mastodon.xml:1]: I think the official word now is `post`, even if some people refuse to drop the `toot` x)

Comment (dasgarner) [reply modules/mastodon.xml:1]: toot toot 😄 

I think post would be better 👍

**Comment by ifarzana** [comment]
We need the appropriate assets for mastodon though.

**Review by dasgarner — COMMENTED** [review]
This looks good, but you're going to have to provide those assets now that they have been renamed.

**Comment by ifarzana** [comment]
@dasgarner this `$dataProvider->getSetting('defaultServerUrl', 'https://mastodon.social')` does not give me data from the settings, do you know what I am missing?

**Review by dasgarner — APPROVED** [review]
This looks good - looking forward to getting it tested.

**Review by dasgarner — APPROVED** [review]

**Review by dasgarner — APPROVED** [review]

**Pull request merged**